scripts/deploy/govcms-db-sync

```diff
@@ -20,30 +21,37 @@ if [ "$LAGOON_ENVIRONMENT_TYPE" = "production" ]; then
   exit 0
 fi
 
+echo "[info]: Environment type: $LAGOON_ENVIRONMENT_TYPE"
+echo "[info]: Content strategy: $GOVCMS_DEPLOY_WORKFLOW_CONTENT"
+echo "[info]: Site alias:       $GOVCMS_SITE_ALIAS"
+echo "[info]: Alias path:       $GOVCMS_SITE_ALIAS_PATH"
+
 echo "[info]: Check that the site can be bootstrapped."
-echo "[info]: Alias path $GOVCMS_SITE_ALIAS_PATH"
-echo "[info]: Site alias $GOVCMS_SITE_ALIAS"
 
 if ! drush status --fields=bootstrap | grep -q "Successful"; then
   # This will ensure that the database is synchronised on the first deploy
-  # of an environment in lagoon.
+  # of an environment in Lagoon.
   echo "[info]: Site could not be bootstrapped... syncing."
-  drush --alias-path="$GOVCMS_SITE_ALIAS_PATH" sql-sync @"$GOVCMS_SITE_ALIAS" @self -y
-  exit 0
-fi
-
-if [ "$GOVCMS_DEPLOY_WORKFLOW_CONTENT" != "import" ]; then
-  # Allow per environment overrides for the synchronisation flow, if the site
+elif [ "$GOVCMS_DEPLOY_WORKFLOW_CONTENT" != "import" ]; then
+  # Allow per environment overrides for the synchronisation flow: if the site
   # has bootstrapped but the environment is not set to "import" for the workflow
   # then we will not synchronise the database.
-  echo "[skip]: Site can be bootstrapped and the workflow is not set to import."
+  echo "[skip]: Site can be bootstrapped and the content workflow is not set to \"import\"."
   exit 0
 fi
 
-echo "[info]: Environment type: $LAGOON_ENVIRONMENT_TYPE"
 echo "[info]: Preparing database sync"
 
-drush --alias-path="$GOVCMS_SITE_ALIAS_PATH" sql-sync @"$GOVCMS_SITE_ALIAS" @self -y
+# Database options to configure the remote to use the read replica when
+# performing read operations.
+sync_opts=""
+if [ -n "$MARIADB_READREPLICA_HOSTS" ] && tables=$(drush @"$GOVCMS_SITE_ALIAS" sqlq 'show tables;' --database=read 2> /dev/null) && [ -n "$tables" ]; then
+  echo "[info]: Replica is available, using for database operations."
+  sync_opts="--source-database=read"
+fi
+
+# shellcheck disable=SC2086
+drush --alias-path="$GOVCMS_SITE_ALIAS_PATH" $sync_opts sql:sync @"$GOVCMS_SITE_ALIAS" @self -y
```

Comment:
note the change from `sql-sync` to `sql:sync`. This is to make sure that we are only supporting Drush 9. `govcms-deploy` already has this change. Moreover, I've tested what is the global Drush version resolution in `govcms8lagoon` image and it is Drush 9.

Reply:
Awesome! Thanks!